src/Sarif.Viewer.VisualStudio.Core/SarifViewerPackage.cs

```diff
@@ -265,10 +297,136 @@ private void SolutionEvents_OnAfterBackgroundSolutionLoadComplete(object sender,
+                            menuCommands.Add(flyoutMenu);
+
+                            return flyoutMenu;
+                        }
+
+                        OleMenuCommand CreateDynamicMenuCommand(ErrorListMenuCommand command, int id)
+                        {
+                            var commandId = new CommandID(ErrorListCommand.ResultSourceServiceCommandSet, id);
+                            var menuCommand = new OleMenuCommand(
+                                (sender, e) => this.InvokeResultSourceServiceMenuCommand(sender),
+                                null, // changeHandler
+                                (sender, e) => BeforeQueryStatusResultSourceServiceMenuItem(sender, command),
+                                commandId);
+                            menuCommand.Properties.Add(ResultSourcesConstants.ResultSourceServiceMenuCommandInvokeCallbackKey, command.InvokeMenuCommand);
+                            menuCommand.Properties.Add(ResultSourcesConstants.ResultSourceServiceMenuCommandBeforeQueryStatusCallbackKey, command.BeforeQueryStatusMenuCommand);
+                            menuCommands.Add(menuCommand);
+
+                            return menuCommand;
+                        }
+
+                        using (OleMenuCommandService mcs = this.GetService<IMenuCommandService, OleMenuCommandService>())
+                        {
+                            int firstMenuId = requestAddMenuCommandEventArgs.FirstMenuId;
+                            int firstCommandId = requestAddMenuCommandEventArgs.FirstCommandId;
+                            int flyoutCount = 0;
+                            int commandCount = 0;
+
+                            for (int f = 0; f < requestAddMenuCommandEventArgs.MenuItems.Flyouts.Count && f < ResultSourceHost.ErrorListContextdMenuChildFlyoutsPerFlyout; f++)
+                            {
+                                ErrorListMenuFlyout flyout = requestAddMenuCommandEventArgs.MenuItems.Flyouts[f];
+                                OleMenuCommand flyoutMenu = CreateFlyoutMenu(flyout, firstMenuId + flyoutCount);
+                                mcs.AddCommand(flyoutMenu);
+                                flyoutCount++;
+
+                                for (int c = 0; c < flyout.Commands.Count && c < ResultSourceHost.ErrorListContextdMenuCommandsPerFlyout; c++)
+                                {
+                                    ErrorListMenuCommand command = flyout.Commands[c];
+                                    OleMenuCommand menuCommand = CreateDynamicMenuCommand(command, firstCommandId + commandCount);
+                                    mcs.AddCommand(menuCommand);
+                                    commandCount++;
+                                }
+                            }
+                        }
+                    }
+
+                    break;
+            }
+        }
+
+        private void InvokeResultSourceServiceMenuCommand(object sender)
```

Comment:
>InvokeResultSourceServiceMenuCommand [](http://example.com/codeflow?start=21&length=36)

nit: menu command related code can be in a separate class e.g. ErrorListCommand.cs

Reply:
Good point.